File: src/Entity/Vehicule.php
<?php

namespace App\Entity;
use App\Repository\VehiculeRepository;
use App\Entity\Categorie;
use Symfony\Component\Validator\Constraints as Assert;




use Doctrine\ORM\Mapping as ORM;



 #[ORM\Entity(repositoryClass: VehiculeRepository::class)]

class Vehicule
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $idVehicule = null ;
   

    
    #[ORM\Column(length:50)]
    #[Assert\NotBlank(message: "Remplir vos champs")]
    #[Assert\Regex(
        pattern: '/^([0-9]+TUN[0-9]+)$/',
        message: "Le nom du livreur doit contenir des caractères alphabétiques uniquement."
    )]
    private ?string $matricule = null ;

    
    #[ORM\Column(length:50)]
    #[Assert\NotBlank(message: "Remplir vos champs")]
    #[Assert\Length(min: 2)]
    private ?string $marque = null ;

    #[ORM\ManyToOne(targetEntity: Categorie::class)]
    #[ORM\JoinColumn(name: "id_categorie", referencedColumnName: "id_categorie")]
    protected $idCategorie;

    public function getIdVehicule(): ?int
    {
        return $this->idVehicule;
    }

    public function getMatricule(): ?string
    {
        return $this->matricule;
    }

    public function setMatricule(string $matricule): self
    {
        $this->matricule = $matricule;

        return $this;
    }

    public function getMarque(): ?string
    {
        return $this->marque;
    }

    public function setMarque(string $marque): self
    {
        $this->marque = $marque;

        return $this;
    }

    public function getIdCategorie(): ?Categorie
    {
        return $this->idCategorie;
    }

    public function setIdCategorie(?Categorie $idCategorie): self
    {
        $this->idCategorie = $idCategorie;

        return $this;
    }


}

File: src/com/mycompany/myapp/entities/Vehicule.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.myapp.entities;

/**
 *
 * @author hp
 */
public class Vehicule {
    
    int id_vehicule ; 
    String matricule , marque ; 
    int id_categorie ;

    public Vehicule() {
    }

    public Vehicule(String matricule, String marque, int id_categorie) {
        this.matricule = matricule;
        this.marque = marque;
        this.id_categorie = id_categorie;
    }

    public Vehicule(int id_vehicule, String matricule, String marque, int id_categorie) {
        this.id_vehicule = id_vehicule;
        this.matricule = matricule;
        this.marque = marque;
        this.id_categorie = id_categorie;
    }

    public int getId_vehicule() {
        return id_vehicule;
    }

    public String getMatricule() {
        return matricule;
    }

    public String getMarque() {
        return marque;
    }

    public int getId_categorie() {
        return id_categorie;
    }

    public void setId_vehicule(int id_vehicule) {
        this.id_vehicule = id_vehicule;
    }

    public void setMatricule(String matricule) {
        this.matricule = matricule;
    }

    public void setMarque(String marque) {
        this.marque = marque;
    }

    public void setId_categorie(int id_categorie) {
        this.id_categorie = id_categorie;
    }

    @Override
    public String toString() {
        return "Vehicule{" + "id_vehicule=" + id_vehicule + ", matricule=" + matricule + ", marque=" + marque + ", id_categorie=" + id_categorie + '}';
    }
    
    
    
}
